apps/webapp/src/clients/penumbra-port.ts

```diff
@@ -0,0 +1,31 @@
+import type {
+  InitChannelClientDataType,
+  InitChannelClientMessage,
+} from '@penumbra-zone/transport/src/types';
+
+import type { Exposed } from '@penumbra-zone/types/src/penumbra-global';
+
+const penumbra = Symbol.for('penumbra');
+
+declare global {
+  interface Window {
+    [penumbra]?: Exposed;
+  }
+}
+
+export const getPenumbraPort = (serviceTypeName: string) => {
+  const { port1: port, port2: transferPort } = new MessageChannel();
+  if (!(penumbra in window)) throw Error('No Penumbra global (extension not installed)');
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
```

Comment:
Think this eslint flag is right. Because of the check above, we can just do:

```typescript
const initPort = window[penumbra].services?.[serviceTypeName];
```